--- profiles/commons/commons.install	2015-02-19 22:29:09.000000000 +0100
+++ profiles/commons/commons.install	2015-06-01 14:14:47.317117475 +0200
@@ -273,7 +273,7 @@
     '#type' => 'checkbox',
     '#title' => st('Install example content'),
     '#description' => st('Install Commons with example content so that you can get a sense of what your site will look like once it becomes more active. Example content includes a group, a few users and content for that group. Example content can be modified or deleted like normal content.'),
-    '#default_value' => TRUE
+    '#default_value' => FALSE
   );
 
   $form['commons_enable_og_access'] = array(
@@ -559,7 +559,7 @@
  */
 function commons_install() {
   // Enable the Origins theme and set it as the default.
-  theme_enable(array('commons_origins'));
+  theme_enable(array('commons_origins', 'community', 'backend'));
 
   // The Bartik theme is automatically enabled during installation. Disable it.
   db_update('system')
@@ -569,10 +569,10 @@
     ->execute();
 
   // Set Commons Origins as the default theme.
-  variable_set('theme_default', 'commons_origins');
+  variable_set('theme_default', 'community');
 
   // Set Ember as the administration theme.
-  variable_set('admin_theme', 'ember');
+  variable_set('admin_theme', 'backend');
 
   // Do not use the administration theme when editing or creating content.
   variable_set('node_admin_theme', FALSE);
@@ -584,7 +584,7 @@
   $menu_block = array(
     'module' => 'system',
     'delta' => 'main-menu',
-    'theme' => 'commons_origins',
+    'theme' => 'community',
     'visibility' => 0,
     'region' => 'menu_bar',
     'status' => 1,
@@ -596,7 +596,7 @@
   $search_block = array(
     'module' => 'search',
     'delta' => 'form',
-    'theme' => 'commons_origins',
+    'theme' => 'community',
     'visibility' => 0,
     'region' => 'header',
     'status' => 1,
