diff --git a/profiles/commons/commons.info b/profiles/commons/commons.info
index c58d29d..b2209a6 100644
--- a/profiles/commons/commons.info
+++ b/profiles/commons/commons.info
@@ -18,7 +18,6 @@ dependencies[] = number
 dependencies[] = options
 dependencies[] = path
 dependencies[] = rdf
-dependencies[] = search
 dependencies[] = shortcut
 dependencies[] = taxonomy
 
@@ -28,7 +27,6 @@ dependencies[] = addressfield_tokens
 dependencies[] = breakpoints
 dependencies[] = ckeditor
 dependencies[] = ctools
-dependencies[] = custom_search
 dependencies[] = dialog
 dependencies[] = diff
 dependencies[] = elements
@@ -57,7 +55,9 @@ dependencies[] = media_oembed
 dependencies[] = og
 dependencies[] = panelizer
 dependencies[] = panels
-dependencies[] = panels_ipe
+dependencies[] = oc_view_mode
+dependencies[] = og_access
+dependencies[] = search_api_db
 dependencies[] = paranoia
 dependencies[] = pathauto
 dependencies[] = privatemsg
@@ -109,8 +109,6 @@ dependencies[] = commons_q_a
 dependencies[] = commons_q_a_pages
 dependencies[] = commons_radioactivity
 dependencies[] = commons_radioactivity_groups
-dependencies[] = commons_search
-dependencies[] = commons_search_core
 dependencies[] = commons_site_homepage
 dependencies[] = commons_social_sharing
 dependencies[] = commons_trusted_contacts
