--- includes/entity.inc	2015-09-07 11:20:37.277111562 +0200
+++ includes/entity.inc	2015-09-07 11:19:11.000000000 +0200
@@ -53,6 +53,9 @@
    * Set up the object instance on construction or unserializiation.
    */
   protected function setUp() {
+    if (!function_exists ('entity_get_info')) {
+       return;
+    }
     $this->entityInfo = entity_get_info($this->entityType);
     $this->idKey = $this->entityInfo['entity keys']['id'];
     $this->nameKey = isset($this->entityInfo['entity keys']['name']) ? $this->entityInfo['entity keys']['name'] : $this->idKey;
@@ -300,6 +303,9 @@
     unset($vars['entityInfo'], $vars['idKey'], $vars['nameKey'], $vars['statusKey']);
     // Also key the returned array with the variable names so the method may
     // be easily overridden and customized.
+    if (!function_exists ('drupal_map_assoc')) {
+      return array();
+    }
     return drupal_map_assoc(array_keys($vars));
   }
 
