--- file_entity.pages_1.inc	2016-04-07 09:18:57.000000000 +0200
+++ file_entity.pages.inc	2016-04-07 09:19:12.334761894 +0200
@@ -841,7 +841,7 @@
   }
 
   if (file_uri_scheme($file->uri) != $form_state['values']['scheme']) {
-    $file_destination = $form_state['storage']['scheme'] . '://' . file_uri_target($file->uri);
+    $file_destination = $form_state['values']['scheme'] . '://' . file_uri_target($file->uri);
     $file_destination = file_stream_wrapper_uri_normalize($file_destination);
     if ($moved_file = file_move($file, $file_destination, FILE_EXISTS_RENAME)) {
       // Only re-assign the file object if file_move() did not fail.
