--- legal.module	2015-10-05 14:35:08.000000000 +0200
+++ legal.module	2015-10-05 14:35:20.920839293 +0200
@@ -426,8 +426,8 @@
   $uid = $user->uid;
 
   // Log the user out and regenerate the Drupal session.
-  module_invoke_all('user_logout', $user);
-  drupal_session_regenerate();
+//  module_invoke_all('user_logout', $user);
+//  drupal_session_regenerate();
 
   // We have to use $GLOBALS to unset a global variable.
   $user = drupal_anonymous_user();
