diff --git a/openid_connect_sso_client.module b/openid_connect_sso_client.module
index cbe20d2..ed2a045 100644
--- a/openid_connect_sso_client.module
+++ b/openid_connect_sso_client.module
@@ -54,8 +54,9 @@ function openid_connect_sso_client_init() {
 function openid_connect_sso_client_menu_alter(&$items) {
   // Don't allow users to edit their accounts, that needs to happen
   // on the server.
-  $items['user/%user/edit']['access callback'] = 'user_access';
-  $items['user/%user/edit']['access arguments'] = array('administer users');
+// Patch Trinoco
+//  $items['user/%user/edit']['access callback'] = 'user_access';
+//  $items['user/%user/edit']['access arguments'] = array('administer users');
 }
 
 /**
