--- openid_connect_1.module	2016-03-21 15:09:32.000000000 +0100
+++ openid_connect.module	2016-03-21 15:10:38.784983801 +0100
@@ -818,10 +818,10 @@
     openid_connect_save_userinfo($account, $userinfo);
   }
 
-  openid_connect_login_user($account, $destination);
-
   module_invoke_all('openid_connect_post_authorize', $tokens, $account, $userinfo, $client->getName());
 
+  openid_connect_login_user($account, $destination);
+  
   return TRUE;
 }
 
