--- profiles/commons/modules/contrib/og/includes/og.field.inc	2015-10-06 11:42:01.000000000 +0200
+++ profiles/commons/modules/contrib/og/includes/og.field.inc	2015-10-06 11:43:59.017659641 +0200
@@ -56,8 +56,7 @@
   // to keep even after saving.
   $element['#other_groups_ids'] = array();
   $element['#element_validate'][] = 'og_complex_widget_element_validate';
-
-  if (user_access('administer group')) {
+  if (user_access('administer group') && $element['#field_name'] != 'og_user_group_ref') {
     $has_admin = TRUE;
     $field_modes[] = 'admin';
   }
