--- template.php	2015-01-21 11:48:02.841577863 +0100
+++ template.php	2015-01-21 11:47:14.000000000 +0100
@@ -243,13 +243,16 @@
 
   // We want to add some basic template suggestions, in case we are using
   // different view modes within our user display.
-  $view_mode = $vars['elements']['#view_mode'];
-  $user = $vars['elements']['#account'];
+  if (isset($vars['elements']['#view_mode']) && isset($vars['elements']['#account'])) {
+    $view_mode = $vars['elements']['#view_mode'];
+    $user = $vars['elements']['#account'];
+
+    $vars['theme_hook_suggestions'][] = 'user_profile';
+    $vars['theme_hook_suggestions'][] = 'user_profile__' . $view_mode;
+    $vars['theme_hook_suggestions'][] = 'user_profile__' . $user->uid;
+    $vars['theme_hook_suggestions'][] = 'user_profile__' . $view_mode . '__' . $user->uid;
+  }
 
-  $vars['theme_hook_suggestions'][] = 'user_profile';
-  $vars['theme_hook_suggestions'][] = 'user_profile__' . $view_mode;
-  $vars['theme_hook_suggestions'][] = 'user_profile__' . $user->uid;
-  $vars['theme_hook_suggestions'][] = 'user_profile__' . $view_mode . '__' . $user->uid;
 }
 
 /**
